### PR TITLE
Add Azure Content Safety java

### PR DESCRIPTION
**Pull request by zhaiyutong**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+System.out.println("Hate severity: " + response.getHateResult().getSeverity());
+System.out.println("SelfHarm severity: " + response.getSelfHarmResult().getSeverity());
+System.out.println("Sexual severity: " + response.getSexualResult().getSeverity());
+System.out.println("Violence severity: " + response.getViolenceResult().getSeverity());
+```
+
+### Manage text blocklist
+
+#### Create or update text blocklist
+```java com.azure.ai.contentsafety.createorupdatetextblocklist
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:199]: maybe just use a simpler codesnippet tag

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+String endpoint = Configuration.getGlobalConfiguration().get("CONTENT_SAFETY_ENDPOINT");
+String key = Configuration.getGlobalConfiguration().get("CONTENT_SAFETY_KEY");
+
+ContentSafetyClient contentSafetyClient = new ContentSafetyClientBuilder()
+    .credential(new AzureKeyCredential(key))
+    .endpoint(endpoint).buildClient();
+
+ImageData image = new ImageData();
+String cwd = System.getProperty("user.dir");
+String absolutePath = cwd + "./src/image.jpg";
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:174]: resource should be under subfolder of samples or test folder (definitely not directly in src folder).

Prefer to be under `src/test/resources`
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileShareTestBase.java#L95

@srnagar Do we have good example of loading file in sample? storage-file-share probably not a good example https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-share/src/samples/java/com/azure/storage/file/share/FileSample.java#L93-L94

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+try {
+    image.setContent(Files.readAllBytes(Paths.get(absolutePath)));
+} catch (IOException e) {
+    throw new RuntimeException(e);
+}
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:179]: Do not teach user do this.

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+System.out.println("Sexual severity: " + response.getSexualResult().getSeverity());
+System.out.println("Violence severity: " + response.getViolenceResult().getSeverity());
+```
+
+### Manage text blocklist
+
+#### Create or update text blocklist
+```java com.azure.ai.contentsafety.createtextblocklist
+String blocklistName = "TestBlocklist";
+BinaryData resource = BinaryData.fromString("{\"description\":\"Test Blocklist\"}");
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:201]: Better to use a Map for payload, and BinaryData.fromObject

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+System.out.println("\nGet BlockItem:");
+System.out.println("BlockItemId: " + getBlockItem.getBlockItemId() + ", Text: " + getBlockItem.getText() + ", Description: " + getBlockItem.getDescription());
+```
+#### Remove blockItems
+```java com.azure.ai.contentsafety.removeblockitems
+String removeBlockItemId = addedBlockItems.getValue().get(0).getBlockItemId();
+List<String> removeBlockItemIds = new ArrayList<>();
+removeBlockItemIds.add(removeBlockItemId);
+contentSafetyClient.removeBlockItems(blocklistName, new RemoveBlockItemsOptions(removeBlockItemIds));
+```
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:262]: sample should be independent. `addedBlockItems` does not belong here.

File: sdk/contentsafety/azure-ai-contentsafety/tsp-location.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+directory: specification/cognitiveservices/ContentSafety
+repo: Azure/azure-rest-api-specs
+commit: 3f4406ada13340d19f77b6ce5b229527b5221008
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/tsp-location.yaml:3]: make sure you use commit id from spec main branch

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+For details on contributing to this repository, see the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).
+
+1. Fork it
+1. Create your feature branch (`git checkout -b my-new-feature`)
+1. Commit your changes (`git commit -am 'Add some feature'`)
+1. Push to the branch (`git push origin my-new-feature`)
+1. Create new Pull Request
+
+<!-- LINKS -->
+[product_documentation]: https://azure.microsoft.com/services/
```
[the remark this answers is not in the document]
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md]: updated

File: sdk/contentsafety/azure-ai-contentsafety/README.md (hunk 4 above)
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md]: updated

File: sdk/contentsafety/azure-ai-contentsafety/README.md (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/contentsafety/azure-ai-contentsafety/README.md:168]: Wait for Srikanta's comment.

My opinion is that in test it should be `getClassLoader().getResource()`.

Not sure about sample. But if we use file, one should use `Paths.get()` to concat file path.

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+## Examples
+The following section provides several code snippets covering some of the most common Content Safety service tasks, including:
+
+* [Analyze text](#analyze-text)
+* [Analyze image](#analyze-image)
+* [Manage text blocklist](#manage-text-blocklist)
+
+Please refer to [sample data](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/contentsafety/Azure.AI.ContentSafety/tests/Samples/sample_data) for the data used here. For more samples, please refer to [Samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/contentsafety/Azure.AI.ContentSafety/tests/Samples).
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:113]: clean up those dotnet reference.

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
+# Azure ContentSafety client library for Java
+
+[Azure AI Content Safety][contentsafety_overview] detects harmful user-generated and AI-generated content in applications and services. Content Safety includes several APIs that allow you to detect material that is harmful:
+
+* Text Analysis API: Scans text for sexual content, violence, hate, and self harm with multi-severity levels.
```
Comment (haolingdong-msft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:5]: It is mentioned that "Content Safety includes several APIs that allow you to detect material that is harmful" but only 3 APIs are mentioned in the list. Make sure all the API descriptions are included in the list.

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+String key = Configuration.getGlobalConfiguration().get("CONTENT_SAFETY_KEY");
+
+ContentSafetyClient contentSafetyClient = new ContentSafetyClientBuilder()
+    .credential(new AzureKeyCredential(key))
+    .endpoint(endpoint).buildClient();
+
+ImageData image = new ImageData();
+String cwd = System.getProperty("user.dir");
+String absolutePath = cwd + "./src/image.jpg";
+image.setContent(Files.readAllBytes(Paths.get(absolutePath)));
```
Comment (haolingdong-msft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:163]: The method "readAllBytes()" can throw an IOException which is not handled. It is recommended to handle the exception.

File: sdk/contentsafety/azure-ai-contentsafety/README.md (hunk 9 above)
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md:5]: we describe all blocklist APIs in Text Blocklist Management APIs
It is aligned with other three languages.

File: sdk/contentsafety/azure-ai-contentsafety/README.md (hunk 10 above)
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md]: Added it into the function public void testAnalyzeImageTests() throws IOException
it is aligned with weidong before.

File: sdk/contentsafety/azure-ai-contentsafety/src/test/java/com/azure/ai/contentsafety/AnalyzeImageTests.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public final class AnalyzeImageTests extends ContentSafetyClientTestBase {
+    @Test
+    public void testAnalyzeImageTests() throws IOException {
+        // method invocation
+        ImageData image = new ImageData();
+        String cwd = System.getProperty("user.dir");
+        String source = "/src/test/resources/image.jpg";
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/src/test/java/com/azure/ai/contentsafety/AnalyzeImageTests.java:21]: If it is just same file, please keep only one. You can load it from samples folder.

File: sdk/contentsafety/azure-ai-contentsafety/CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
+
+- Azure ContentSafety client library for Java. This package contains Microsoft Azure ContentSafety client library.
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/CHANGELOG.md:5]: Please update this line, to best describe the module.

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+az cognitiveservices account show --name "resource-name" --resource-group "resource-group-name" --query "properties.endpoint"
+```
+
+#### Get the API key
+
+The API key can be found in the [Azure Portal][azure_portal] or by running the following [Azure CLI][azure_cli_key_lookup] command:
+
+```bash
+az cognitiveservices account keys list --name "<resource-name>" --resource-group "<resource-group-name>"
+```
```
Comment (weidongxu-microsoft) [inline sdk/contentsafety/azure-ai-contentsafety/README.md:52]: nit, you may want to make cmd consistent.

one is name with out `<>`, another is with `<>`

Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md]: resolved

**Review by joshfree — APPROVED** [review]

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 14]
```diff
@@ -0,0 +1,288 @@
+# Azure ContentSafety client library for Java
+
+[Azure AI Content Safety][contentsafety_overview] detects harmful user-generated and AI-generated content in applications and services. Content Safety includes several APIs that allow you to detect material that is harmful:
```
[the remark this answers is not in the document]
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md:3]: Thank you for your comment, we prefer to keep this word which is aligned with https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/contentsafety/Azure.AI.ContentSafety

File: sdk/contentsafety/azure-ai-contentsafety/README.md [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+String removeBlockItemId = addedBlockItems.getValue().get(0).getBlockItemId();
+List<String> removeBlockItemIds = new ArrayList<>();
+removeBlockItemIds.add(removeBlockItemId);
+contentSafetyClient.removeBlockItems(blocklistName, new RemoveBlockItemsOptions(removeBlockItemIds));
+```
+#### Delete text blocklist
+```java com.azure.ai.contentsafety.deletetextblocklist
+contentSafetyClient.deleteTextBlocklist(blocklistName);
+```
+## Troubleshooting
```
[the remark this answers is not in the document]
Comment (zhaiyutong) [reply sdk/contentsafety/azure-ai-contentsafety/README.md:248]: Thanks for this comment, we will consider add a TROUBLESHOOTING.md in GA version.

File: sdk/contentsafety/azure-ai-contentsafety/README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/contentsafety/azure-ai-contentsafety/README.md:3]: Before GA, you may want to check again, to see whether you can sync this for all languages.

**Pull request merged**